sllm/serve/routers/migration_router.py

```diff
@@ -0,0 +1,239 @@
+        source_instance = migration_plan.source_instance
+        source_instance_id = source_instance.instance_id
+        target_node_id = migration_plan.target_node_id
+        # start the target_instance on the target node
+        startup_config = {
+            "num_cpus": self.resource_requirements["num_cpus"],
+            "num_gpus": self.resource_requirements["num_gpus"],
+            "resources": {
+                "worker_node": 0.1,
+                f"worker_id_{target_node_id}": 0.1,
+            },
+        }
+        logger.info(f"Startup config: {startup_config}, {self.backend_config}")
+
+        instance_id = self._new_instance_id()
+        logger.info(
+            f"Creating new instance {instance_id} for model {self.model_name}"
+        )
+        # TODO: Add max_queue_length to target_instance
+        target_instance = InstanceHandle(
+            instance_id=instance_id,
+            max_queue_length=10,
+            num_gpu=self.resource_requirements["num_gpus"],
+        )
+
+        ret = await self.model_loading_scheduler.mark_resource.remote(
+            self.model_name, instance_id, target_node_id
+        )
+        if not ret:
+            logger.error(
+                f"Failed to mark resource for instance {instance_id} for model {self.model_name}"
+            )
+            return None
+
+        await start_instance.options(
+            resources={
+                "worker_node": 0.1,
+                f"worker_id_{target_node_id}": 0.1,
+            }
+        ).remote(
+            instance_id,
+            self.backend,
+            self.model_name,
+            self.backend_config,
+            startup_config,
+        )
+        logger.info(
+            f"Started instance {instance_id} for model {self.model_name}"
+        )
+        target_instance.backend_instance = ray.get_actor(instance_id)
+        async with target_instance.lock:
+            target_instance.ready = True
+            target_instance.node_id = target_node_id
+        logger.info(
+            f"Initialized instance {instance_id} for model {self.model_name}"
+        )
+        await target_instance.backend_instance.init_backend.remote()
+        logger.info(
+            f"Initialized backend for instance {instance_id} for model {self.model_name}"
+        )
+        # migrate the tokens from the source_instance (if still running) to the target_instance
+        if source_instance_id not in self.ready_instances:
+            logger.info(f"Instance {source_instance_id} not found")
+            target_instance.ready = False
+            await target_instance.backend_instance.shutdown.remote()
+            ray.kill(target_instance.backend_instance)
+            return None
+        source_instance = self.ready_instances[source_instance_id]
+        migration_iter = 0
+        n_previous_tokens = 0
+        while True:
+            logger.info(f"Migration iteration {migration_iter}")
+            current_tokens = ray.get(
```

Comment:
It is okay for now, but since we're going to support vllm backend in the future as well, and vllm has continuous batching, I think `get_current_tokens()` at that time should return a List[List[int]], where each element in the outer List represents each request in the batch, and each element in the inner List represents a token in that request

Reply:
Fixed.